futures-core/src/future/mod.rs

```diff
@@ -12,158 +11,114 @@ mod option;
+/// implements lightweight threads in user space.
 ///
-/// More information about the details of `poll` and the nitty-gritty of tasks
-/// can be [found online at tokio.rs][poll-dox].
-///
-/// [poll-dox]: https://tokio.rs/docs/going-deeper-futures/futures-model/
+/// When using a future, you generally won't call `poll` directly, but instead
+/// use combinators to build up asynchronous computations. A complete
+/// computation can then be spawned onto an [executor](::executor::Executor) as
+/// a new, independent task that will automatically be `poll`ed to completion.
 ///
 /// # Combinators
 ///
 /// Like iterators, futures provide a large number of combinators to work with
 /// futures to express computations in a much more natural method than
-/// scheduling a number of callbacks. For example the `map` method can change
-/// a `Future<Item=T>` to a `Future<Item=U>` or an `and_then` combinator could
-/// create a future after the first one is done and only be resolved when the
-/// second is done.
-///
-/// Combinators act very similarly to the methods on the `Iterator` trait itself
-/// or those on `Option` and `Result`. Like with iterators, the combinators are
-/// zero-cost and don't impose any extra layers of indirection you wouldn't
-/// otherwise have to write down.
-///
-/// More information about combinators can be found [on tokio.rs].
-///
-/// [on tokio.rs]: https://tokio.rs/docs/going-deeper-futures/futures-mechanics/
+/// scheduling a number of callbacks. As with iterators, the combinators are
+/// zero-cost: they compile away. You can find the combinators in the
+/// [future-util](https://docs.rs/futures-util) crate.
 pub trait Future {
-    /// The type of value that this future will resolved with if it is
-    /// successful.
+    /// A successful value
     type Item;
 
-    /// The type of error that this future will resolve with if it fails in a
-    /// normal fashion.
+    /// An error
     type Error;
 
-    /// Query this future to see if its value has become available, registering
-    /// interest if it is not.
-    ///
-    /// This function will check the internal state of the future and assess
-    /// whether the value is ready to be produced. Implementers of this function
-    /// should ensure that a call to this **never blocks** as event loops may
-    /// not work properly otherwise.
-    ///
-    /// When a future is not ready yet, the `Async::Pending` value will be
-    /// returned. In this situation the future will *also* register interest of
-    /// the current task in the value being produced. This is done by calling
-    /// `task::park` to retrieve a handle to the current `Task`. When the future
-    /// is then ready to make progress (e.g. it should be `poll`ed again) the
-    /// `unpark` method is called on the `Task`.
+    /// Attempt to resolve the future to a final value, registering
+    /// the current task for wakeup if the value is not yet available.
```

Comment:
ditto about "current task"-- it seems like `cx.waker()` should be mentioned directly.